ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs

```diff
@@ -84,17 +84,21 @@ impl IndexerStream for DatastreamServer {
         req: Request<RawDatastreamRequest>,
     ) -> Result<Response<Self::RawDatastreamStream>, Status> {
         // Get request identity. The request is already authenticated by the interceptor.
-        let request_token = req
+        let request_token = match req
             .metadata()
             .get(GRPC_AUTH_TOKEN_HEADER)
-            .unwrap()
-            .to_str()
-            .unwrap()
-            .to_string();
+            .map(|token| token.to_str().unwrap_err().to_string())
```

Comment:
I might be missing this, but doesn't this always unwrap fail if the `token.to_str()` works successfully?

Reply:
going to hijack this PR; but what i am going to do is:

```
let request_token = match req
            .metadata()
            .get(GRPC_AUTH_TOKEN_HEADER)
            .map(|token| token.to_str())
        {
            Some(Ok(token)) => token.to_string(),
            // It's required to have a valid request token.
            _ => return Result::Err(Status::aborted("Invalid request token")),
        };

```